File: benchmarks/python/file_stats.py
from pathlib import Path


def main() -> None:
    lines = ["the quick brown fox", "jumps over", "the lazy dog"]
    path = Path("parley_demo.txt")
    path.write_text("\n".join(lines), encoding="utf-8")

    content = path.read_text(encoding="utf-8")
    rows = content.split("\n")
    print(f"the file has {len(rows)} lines")
    for row in rows:
        if "lazy" in row:
            print(f"found it: {row}")


if __name__ == "__main__":
    main()

File: benchmarks/rust/file_stats.rs
use std::fs;

fn main() {
    let lines = ["the quick brown fox", "jumps over", "the lazy dog"];
    fs::write("parley_demo.txt", lines.join("\n")).expect("write file");

    let content = fs::read_to_string("parley_demo.txt").expect("read file");
    let rows: Vec<&str> = content.split('\n').collect();
    println!("the file has {} lines", rows.len());
    for row in rows {
        if row.contains("lazy") {
            println!("found it: {}", row);
        }
    }
}
